### PR TITLE
Create file - card definition

### PR DESCRIPTION
**Pull request by habdelra**
This PR introduces the create new file - card definition.

create-card-def

### CODE REVIEW

File: packages/host/tests/acceptance/code-submode/create-file-test.gts [hunk 1]
```diff
@@ -467,4 +461,215 @@ module('Acceptance | code submode | create-file tests', function (hooks) {
...
+    await waitFor('[data-test-card-schema="Test Card"]');
+    assert
+      .dom('[data-test-card-schema]')
+      .exists({ count: 3 }, 'the card hierarchy is displayed in schema editor');
+    assert.dom('[data-test-total-fields]').containsText('3 Fields');
+  });
+
+  skip<TestContextWithSave>('TODO can create a new field definition', async function (_assert) {
+    // TODO include percy snapshot
+  });
```
Comment (backspace) [inline packages/host/tests/acceptance/code-submode/create-file-test.gts:511]: Is this for a future PR?

Comment (jurgenwerk) [reply packages/host/tests/acceptance/code-submode/create-file-test.gts:511]: Yeah, I have a ticket for new field definition opened. Luckily that will be very easy to do after this PR 😁

**Review by jurgenwerk — APPROVED** [review]
Looking good, I also tested it and it works!

**Review by tintinthong — APPROVED** [review]

**Comment by jurgenwerk** [comment]
@habdelra I based https://github.com/cardstack/boxel/pull/865 (field definition) on this branch. If you like we can merge that first into this one and rename this PR to mention both card and field definition.

**Comment by habdelra** [comment]
> @habdelra I based #865 (field definition) on this branch. If you like we can merge that first into this one and rename this PR to mention both card and field definition.

oh whoops, i just saw this message after merging this PR. hopefully the merge conflicts are simple to deal with

**Pull request merged**